packages/run-protocol/src/runStake/runStakeManager.js

```diff
@@ -5,123 +5,156 @@ import { makeRatio } from '@agoric/zoe/src/contractSupport/ratio.js';
  * @param {bigint} timing.startTimeStamp
  *
- * @typedef {ReturnType<typeof makeRunStakeManager>} RunStakeManager
+ * @returns {State}
  */
-export const makeRunStakeManager = (
+const initState = (
   zcf,
   debtMint,
   brands,
-  paramManager,
-  mintAndReallocateWithFee,
-  burnDebt,
-  { timerService, chargingPeriod, recordingPeriod, startTimeStamp },
+  mintPowers,
+  { chargingPeriod, recordingPeriod, startTimeStamp, timerService },
 ) => {
-  /** @param { Amount<'copyBag'>} attestationGiven */
-  const maxDebtForLien = attestationGiven => {
-    const mintingRatio = paramManager.getMintingRatio();
-    assert.equal(
-      mintingRatio.numerator.brand,
-      brands.debt,
-      X`${mintingRatio} not in Debt / Stake`,
-    );
-    assert.equal(
-      mintingRatio.denominator.brand,
-      brands.Stake,
-      X`${mintingRatio} not in Debt / Stake`,
-    );
-    assert.equal(
-      attestationGiven.brand,
-      brands.Attestation,
-      X`Invalid Attestation ${attestationGiven}. Expected brand ${brands.Attestation}`,
-    );
-    fit(attestationGiven.value, M.bagOf([M.string(), M.bigint()]));
-    const [[_addr, valueLiened]] = getCopyBagEntries(attestationGiven.value);
-    const amountLiened = AmountMath.make(brands.Stake, valueLiened);
-    const maxDebt = floorMultiplyBy(amountLiened, mintingRatio);
-    return { maxDebt, amountLiened };
-  };
+  const totalDebt = AmountMath.makeEmpty(brands.debt, 'nat');
+  const compoundedInterest = makeRatio(100n, brands.debt); // starts at 1.0, no interest
+  const latestInterestUpdate = startTimeStamp;
 
-  let totalDebt = AmountMath.makeEmpty(brands.debt, 'nat');
-  let compoundedInterest = makeRatio(100n, brands.debt); // starts at 1.0, no interest
-  let latestInterestUpdate = startTimeStamp;
-  const { updater: assetUpdater, notifier: assetNotifer } = makeNotifierKit(
+  const { updater: assetUpdater, notifier: assetNotifier } = makeNotifierKit(
     harden({
       compoundedInterest,
-      interestRate: paramManager.getInterestRate(),
+      interestRate: mintPowers.getGovernedParams().getInterestRate(),
       latestInterestUpdate,
       totalDebt,
     }),
   );
 
+  // ??? does this promise need to be a presence?
+  const periodNotifier = E(timerService).makeNotifier(0n, recordingPeriod);
+  const { zcfSeat: poolIncrementSeat } = zcf.makeEmptySeatKit();
+
+  return {
+    assetNotifier,
+    assetUpdater,
+    brands,
+    chargingPeriod,
+    compoundedInterest,
+    debtMint,
+    latestInterestUpdate,
+    mintPowers,
+    periodNotifier,
+    poolIncrementSeat,
+    recordingPeriod,
+    startTimeStamp,
+    totalDebt,
+    zcf,
+  };
+};
+
+/**
+ *
+ * @param {MethodContext} context
+ */
+const finish = ({ state }) => {
+  const { debtMint, mintPowers, periodNotifier, poolIncrementSeat, zcf } =
+    state;
+
   /**
    *
    * @param {bigint} updateTime
-   * @param {ZCFSeat} poolIncrementSeat
    */
-  const chargeAllVaults = async (updateTime, poolIncrementSeat) => {
```

Comment:
Where did `poolIncrementSeat` go? I feel like I missed a memo.

Reply:
Now that `chargeAllVaults` is defined within `finish()` it's in scope on line 111.

Reply:
Trading diffs is evidently not sufficient here. I should check out the branch and look locally.